pkg/testcase/upgradesuc.go

```diff
@@ -0,0 +1,56 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/rancher/distros-test-framework/pkg/assert"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+// TestUpgradeClusterSUC upgrades cluster using the system-upgrade-controller.
+func TestUpgradeClusterSUC(version string) error {
+	if version == "" {
+		return shared.ReturnLogError("please provide a non-empty version or commit to upgrade to")
+	}
+	shared.PrintClusterState()
+	fmt.Printf("\nUpgrading cluster to: %s\n", version)
+
+	workloadErr := shared.ManageWorkload("apply", "suc.yaml")
+	Expect(workloadErr).NotTo(HaveOccurred(),
+		"system-upgrade-controller manifest did not deploy successfully")
+
+	getPodsSystemUpgrade := "kubectl get pods -n system-upgrade --kubeconfig="
+	err := assert.CheckComponentCmdHost(
+		getPodsSystemUpgrade+shared.KubeConfigFile,
+		"system-upgrade-controller",
+		statusRunning,
+	)
+	Expect(err).NotTo(HaveOccurred(), err)
+
+	product, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred())
+
+	originalFilePath := shared.BasePath() +
+		fmt.Sprintf("/workloads/amd64/%s-upgrade-plan.yaml", product)
+	newFilePath := shared.BasePath() + "/workloads/amd64/plan.yaml"
```

Comment:
This will fail if we want to SUC upgrade arm

Reply:
good question , definetely not , i dont remember if we plan do SUC on ARM, if we want  i can change that
( this file i just copied and paste , separately )

Reply:
We can probably add a TODO to update it once we plan to do SUC on arm. I think we do support SUC upgrade on k3s and rke2 (>=1.27)

Reply:
https://github.com/rancher/distros-test-framework/issues/78